lib/phase/combiner.cpp

```diff
@@ -116,6 +116,10 @@ Result Combiner::dispatch(const FinishTask &t, TaskQueue &tq, int node_id)
   return Result::ONGOING;
 }
 
+void Combiner::mark_init_reached() {
+  phases[0]->mark_init_reached();
```

Comment:
I don't think it is always phase[0]

Reply:
It can happen that `phase[0]` may be some "ephemeral" phase that does nothing and the actual phase is at position 1. Writing this code is error-prone in my opinion.